addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -2583,6 +2584,131 @@ export class OdooEditor extends EventTarget {
     // TABLE MANAGEMENT
     // ================
 
+    /**
+     * Sets selection in table cells in order to select the cells
+     * rectangularly when pressing shift + arrow key (just like
+     * google docs).
+     * 
+     * @private
+     * @param {KeyboardEvent}
+     */
+    _setSelectionInTable(ev) {
+        const selection = this.document.getSelection();
+        let startTable = closestElement(selection.anchorNode, 'table');
+        let endTable = closestElement(selection.focusNode, 'table');
+        if (!startTable || !endTable || !(startTable.contains(endTable) || endTable.contains(startTable))) {
+            return;
+        }
+        let [startTd, endTd] = [closestElement(selection.anchorNode, 'td'), closestElement(selection.focusNode, 'td')];
+        if (startTable !== endTable) {
+            if (endTable.contains(startTable)) {
+                startTd = endTd;
+                startTable = endTable;
+            } else {
+                endTd = startTd;
+                endTable = startTable;
+            }
+        }
+        const endPosition = { x: getRowIndex(endTd), y: getColumnIndex(endTd) };
```

Comment:
what happens if getRowIndex(endTd) or getColumnIndex(endTd) fail? IMO Adding some error handling or checks could make the code more robust.